core-bundle/src/Resources/contao/models/PageModel.php

```diff
@@ -281,6 +281,40 @@ class PageModel extends Model
 	 */
 	protected $blnDetailsLoaded = false;
 
+	public function __get($strKey)
+	{
+		// Lazy-load the fallback language (see contao/core#6874)
+		if ($strKey == 'rootFallbackLanguage' && $this->blnDetailsLoaded && !\array_key_exists('rootFallbackLanguage', $this->arrData))
+		{
+			$this->rootFallbackLanguage = $this->rootIsFallback ? $this->language : null;
+
+			if (!$this->rootIsFallback && ($objFallback = static::findPublishedFallbackByHostname($this->domain)) !== null)
+			{
+				$this->rootFallbackLanguage = $objFallback->language;
+			}
+		}
+
+		return parent::__get($strKey);
+	}
+
+	public function __isset($strKey)
+	{
+		if ($strKey == 'rootFallbackLanguage' && $this->blnDetailsLoaded)
+		{
+			return true;
+		}
+
+		return parent::__isset($strKey);
+	}
+
+	public function row()
+	{
+		// Load lazy properties
+		$this->__get('rootFallbackLanguage');
```

Comment:
I am not sure about this change. Now `rootFallbackLanguage` will always be available even if we have not called `loadDetails()`. And it will return `null` anyway, because `$this->rootIsFallback` will not be set without calling `loadDetails()`, therefore we might as well not preload the value.

Reply:
No, it won't:

```php
 		if ($strKey == 'rootFallbackLanguage' && $this->blnDetailsLoaded && !\array_key_exists('rootFallbackLanguage', $this->arrData))
```

Reply:
True. NVM.